core/shared/src/main/scala/chartreuse/Axes.scala

```diff
@@ -0,0 +1,377 @@
+ */
+
+package chartreuse
+
+import doodle.algebra.*
+import doodle.core.*
+import doodle.syntax.all.*
+
+final case class Axes[-Alg <: Algebra](
+    xTickLayout: MajorTickLayout,
+    yTickLayout: MajorTickLayout,
+    minorTickLayout: MinorTickLayout,
+    grid: Boolean,
+    layers: Seq[Layer[?, Alg]],
+    width: Int,
+    height: Int
+) {
+  private val tickSize = 7
+  private val axisMargin = 10
+  private val textMargin = axisMargin + tickSize + 5
+
+  def build(using
+      numberFormat: NumberFormat
+  ): Picture[Alg & PlotAlg, Unit] = {
+    val dataBoundingBox = layers.foldLeft(BoundingBox.empty) { (bb, layer) =>
+      bb.on(layer.boundingBox)
+    }
+
+    val dataMinX = dataBoundingBox.left
+    val dataMaxX = dataBoundingBox.right
+    val dataMinY = dataBoundingBox.bottom
+    val dataMaxY = dataBoundingBox.top
+
+    val scale = Scale.linear.build(dataBoundingBox, width, height)
+
+    // Convert the Ticks to a sequence of points
+    val asX: Double => Point = x => Point(x, 0)
+    val asY: Double => Point = y => Point(0, y)
+    val xFilter: Double => Boolean = tick =>
+      tick >= dataMinX && tick <= dataMaxX
+    val yFilter: Double => Boolean = tick =>
+      tick >= dataMinY && tick <= dataMaxY
+
+    val xTicksSequence: TicksSequence =
+      majorTickLayoutToSequence(
+        xTickLayout,
+        scale,
+        asX,
+        xFilter,
+        dataMinX,
+        dataMaxX
+      )
+    val yTicksSequence: TicksSequence =
+      majorTickLayoutToSequence(
+        yTickLayout,
+        scale,
+        asY,
+        yFilter,
+        dataMinY,
+        dataMaxY
+      )
+
+    val xMajorTickToMinorTick: (ScreenCoordinate, Double, Int) => (
+        ScreenCoordinate,
+        DataCoordinate
+    ) = (screenCoordinate, interval, i) => {
+      val x = screenCoordinate.x - interval * i
+      (
+        ScreenCoordinate(x, 0),
+        DataCoordinate(x, 0)
+      )
+    }
+
+    val yMajorTickToMinorTick: (ScreenCoordinate, Double, Int) => (
+        ScreenCoordinate,
+        DataCoordinate
+    ) = (screenCoordinate, interval, i) => {
+      val y = screenCoordinate.y - interval * i
+      (
+        ScreenCoordinate(0, y),
+        DataCoordinate(0, y)
+      )
+    }
+
+    val xMinorTicksSequence =
+      if xTickLayout != MajorTickLayout.NoTicks then
+        minorTickLayoutToSequence(
+          minorTickLayout,
+          xTicksSequence,
+          xMajorTickToMinorTick,
+          coordinate => coordinate.x
+        )
+      else List.empty
+    val yMinorTicksSequence =
+      if yTickLayout != MajorTickLayout.NoTicks then
+        minorTickLayoutToSequence(
+          minorTickLayout,
+          yTicksSequence,
+          yMajorTickToMinorTick,
+          coordinate => coordinate.y
+        )
+      else List.empty
+
+    val xTicksMin = Math.min(
+      if xTickLayout != MajorTickLayout.NoTicks then xTicksSequence.head._1.x
+      else Double.MaxValue,
+      scale(Point(dataMinX, 0)).x
+    )
```

Comment:
I know It's better to use pattern matching in such situations, but I used tuple methods just to make the expressions shorter. Let me know if there is a better way to write these expressions.

P.S. Math functions prevent layout issues when we use `Manual` or `NoTicks` layout.